cmake/OpenCVConfig.cmake

```diff
@@ -47,7 +47,7 @@ endif()
 
 if(NOT DEFINED OpenCV_STATIC)
   # look for global setting
-  if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+  if(BUILD_SHARED_LIBS)
```

Comment:
> in case OpenCV was not built with BUILD_SHARED_LIBS=ON

That's incorrect. This is executed in the context of the user's build script, so it checks whether the _user_ has set `BUILD_SHARED_LIBS`.

> So what you want to preserve is the behaviour of the package not being found in this case instead of finding the static libs, correct?

Wouldn't it find the dynamic build, instead?